packages/recommend-react/src/useFrequentlyBoughtTogether.ts

```diff
@@ -17,32 +18,43 @@ export function useFrequentlyBoughtTogether<TObject>({
   recommendClient,
   threshold,
   transformItems: userTransformItems,
-}: GetFrequentlyBoughtTogetherProps<TObject>) {
-  const [result, setResult] = useState<GetRecommendationsResult<TObject>>({
-    recommendations: [],
-  });
+  initialState,
+}: GetFrequentlyBoughtTogetherProps<TObject> & {
+  initialState?: InitialResults<TObject>;
+}) {
+  const initialResults = initialState ?? { recommendations: [] };
+  const [result, setResult] = useState<GetRecommendationsResult<TObject>>(
+    initialResults
+  );
+  const renderRef = useRef(false);
+
   const { status, setStatus } = useStatus('loading');
   const objectIDs = useStableValue(userObjectIDs);
   const transformItems = useStableValue(userTransformItems);
   const queryParameters = useStableValue(userQueryParameters);
 
-  useAlgoliaAgent({ recommendClient });
+  useAlgoliaAgent({ recommendClient, initialState });
 
   useEffect(() => {
-    setStatus('loading');
-    getFrequentlyBoughtTogether({
-      indexName,
-      maxRecommendations,
-      objectIDs,
-      queryParameters,
-      recommendClient,
-      threshold,
-      transformItems,
-    }).then((response) => {
-      setResult(response);
-      setStatus('idle');
-    });
+    if (!initialState || renderRef.current) {
+      setStatus('loading');
+      getFrequentlyBoughtTogether({
+        indexName,
+        maxRecommendations,
+        objectIDs,
+        queryParameters,
+        recommendClient,
+        threshold,
+        transformItems,
+      }).then((response) => {
+        setResult(response);
+        setStatus('idle');
+      });
+    } else {
+      renderRef.current = true;
+    }
   }, [
+    initialState,
```

Comment:
We should stabilize the `initialState` above with `useStableValue()`, otherwise if users inline their `initialState`, it'll trigger an infinite loop.